src/main/java/util/PiazzaLogger.java

```diff
@@ -118,11 +121,54 @@ public void init() {
 	 */
 	public void log(String logMessage, Severity severity) {
 		LoggerPayload loggerPayload = getLoggerPayload();
-		executor.execute(new LogWorker(loggerPayload, logMessage, severity));
+		indexLog(loggerPayload, logMessage, severity); 
 	}
 
 	/**
-	 * Sends an audit message to pz-logger
+	 * Creates elasticsearch index with default mapping.
+	 * 
+	 * @param client
+	 *            elasticsearch client
+	 * @param indexName
+	 *            index to save to
+	 * @param type
+	 *            schema to save to
+	 * @param mapping
+	 *            mapping object
+	 * @return boolean
+	 */
+	public boolean createIndexWithMapping(Client client, String indexName, String type, String mapping) {
+		try {
+			if (!indexExists(indexName)) {
+				CreateIndexRequestBuilder createIndexRequestBuilder = client.admin().indices().prepareCreate(indexName);
+				if (mapping != null) {
+					createIndexRequestBuilder.addMapping(type, mapping);
+				} else {
+					createIndexRequestBuilder.addMapping(type);
+				}
+				CreateIndexResponse response = createIndexRequestBuilder.execute().actionGet();
+				return response.isAcknowledged();
+			}
+		} catch (Exception e) {
+			LOGGER.info(String.format("Unable to create Elasticsearch index %s, it should already exist, error: %s",
+					indexName, e.getMessage()));
```

Comment:
You will need to log the actual exception object. SonarQube will complain otherwise.